### PR TITLE
PoC: First, very very very basic projectview usage

### PR DESCRIPTION
**Pull request by Abrams27**
This is first projectview usage,

currently we are using `targets` section - given targets are included / excluded

This is very early PoC, nevertheless it might help working with huge repos

### CODE REVIEW

File: src/main/java/org/jetbrains/bsp/bazel/server/bsp/services/BuildServerService.java [hunk 1]
```diff
@@ -204,9 +208,13 @@ private SourcesItem createSourcesForLabelAndItemsAndRoots(
 
       throw new RuntimeException("Could not resolve " + fileUri + " within workspace " + prefix);
     }
+    String kindInput =
+        String.format(
+            "rdeps(%s, %s, 1)",
+            TargetsUtils.getAllProjectTargetsWithExcludedTargets(projectView),
+            fileUri.substring(prefix.length()));
```
Comment (andrefmrocha) [inline src/main/java/org/jetbrains/bsp/bazel/server/bsp/services/BuildServerService.java:215]: Maybe this should all be extracted to `TargetsUtils` as a whole?

**Review by andrefmrocha — COMMENTED** [review]
This is looking really good, great work! Some minor comments only

File: src/main/java/org/jetbrains/bsp/bazel/projectview/parser/ProjectViewRawSectionParser.java [hunk 2]
```diff
@@ -19,7 +19,14 @@ private ProjectViewRawSectionParser(ProjectViewSectionParser<T> parser) {
     return new ProjectViewRawSectionParser<>(parser);
   }
 
-  public Optional<T> parseRawSections(List<ProjectViewRawSection> rawSections) {
+  public T parseRawSections(List<ProjectViewRawSection> rawSections) {
+    return parseOptionalRawSections(rawSections)
+        .orElseThrow(
+            () ->
+                new IllegalArgumentException(this.parser.sectionName() + " section is required!"));
+  }
```
[the remark this answers is not in the document]
Comment (Abrams27) [reply src/main/java/org/jetbrains/bsp/bazel/projectview/parser/ProjectViewRawSectionParser.java:27]: I wanted to keep original projetview contract, those two sections are required, rest are optional

File: src/main/java/org/jetbrains/bsp/bazel/projectview/model/ProjectView.java [hunk 3]
```diff
@@ -11,10 +10,10 @@
  */
 public class ProjectView {
 
-  private final Optional<DirectoriesSection> directories;
-  private final Optional<TargetsSection> targets;
+  private final DirectoriesSection directories;
+  private final TargetsSection targets;
 
-  private ProjectView(Optional<DirectoriesSection> directories, Optional<TargetsSection> targets) {
```
[the remark this answers is not in the document]
Comment (Abrams27) [reply src/main/java/org/jetbrains/bsp/bazel/projectview/model/ProjectView.java:17]: I think that those two are a bit different things, I decided to remove `BazelBspServerConfig`, but now I think that maybe I should rollback it and change targets field into projectview. 

Summarizing - `BazelBspServerConfig` with 2 fields - `bazelPath`, `projectView`

File: src/main/java/org/jetbrains/bsp/bazel/projectview/parser/ProjectViewRawSectionParser.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Abrams27) [reply src/main/java/org/jetbrains/bsp/bazel/projectview/parser/ProjectViewRawSectionParser.java:27]: thats also a reason, why i extracted `parseRawSections` and `parseOptionalRawSections` - preparation for further optional sections

[the remark this answers is not in the document]
Comment (andrefmrocha) [reply src/main/java/org/jetbrains/bsp/bazel/projectview/parser/ProjectViewRawSectionParser.java:27]: thanks for explaining it so clearly!

File: src/main/java/org/jetbrains/bsp/bazel/projectview/model/ProjectView.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrefmrocha) [reply src/main/java/org/jetbrains/bsp/bazel/projectview/model/ProjectView.java:17]: agreed!

**Pull request merged**